go/ct/evm_fuzz_test.go

```diff
@@ -0,0 +1,285 @@
+func FuzzDifferentialLfvmVsGeth(f *testing.F) {
+	differentialFuzz(f,
+		lfvm.NewConformanceTestingTarget(),
+		geth.NewConformanceTestingTarget(),
+	)
+}
+
+// TODO: This test makes sense but cannot be enabled yet:
+// - The evmzero fails the differential test against geth. (issue #549)
+// - Any other invocation of fuzzing tests in this file seem to
+// invoke the all other tests in the file, and they will fail.
+// func FuzzDifferentialEvmzeroVsGeth(f *testing.F) {
+// 	differentialFuzz(f,
+// 		evmzero.NewConformanceTestingTarget(),
+// 		geth.NewConformanceTestingTarget(),
+// 	)
+// }
+
+//////////////////////////////////////////////////////////////////////////////
+// Fuzzing helpers
+
+func differentialFuzz(f *testing.F, testeeVm ct.Evm, referenceVm ct.Evm) {
+
+	rnd := rand.New(0)
+
+	prepareFuzzingSeeds(f, rnd)
+
+	f.Fuzz(func(t *testing.T, opCodes []byte, gas int64, revision byte, stackBytes []byte) {
+
+		state, err := corpusInputDataToCtState(opCodes, gas, revision, stackBytes)
+		defer state.Release()
+		if err != nil {
+			t.Skip(err)
+		}
+
+		testeeResultState, err := testeeVm.StepN(state.Clone(), 1)
+		defer func() {
+			if testeeResultState != nil {
+				testeeResultState.Release()
+			}
+		}()
+		if err != nil {
+			t.Fatalf("failed to run test case: %v", err)
+		}
+
+		referenceResultState, err := referenceVm.StepN(state.Clone(), 1)
+		defer func() {
+			if referenceResultState != nil {
+				referenceResultState.Release()
+			}
+		}()
+		if err != nil {
+			t.Fatalf("failed to run test case in reference VM: %v", err)
+		}
+
+		if testeeResultState.Status != referenceResultState.Status {
+			t.Fatal("invalid result, status does not match reference status:", errorReportString(state, testeeResultState, referenceResultState))
+		}
+
+		// if result is other than running, further checks may be misleading
+		// TODO: let Stopped, Returned, and Reverted be diffed as well as their results are more stable than Failed results
+		// This can be done once issue #547 is solved
+		if testeeResultState.Status != st.Running {
+			return
+		}
+
+		if testeeResultState.Gas != referenceResultState.Gas {
+			t.Fatal("invalid result, gas does not match reference gas:", errorReportString(state, testeeResultState, referenceResultState))
+		}
+
+		// Hack: lfvm does a pc transformation, but for code smaller than the required jump the pc will point to a different location
+		// - Geth will point to pc+offset, whenever it is an overflow
+		// - Lfvm will point min(pc+offset, len(code))
+		if testeeResultState.Pc == uint16(len(opCodes)) &&
+			testeeResultState.Pc != referenceResultState.Pc {
+			testeeResultState.Pc = referenceResultState.Pc
+		}
+
+		if !testeeResultState.Eq(referenceResultState) {
+			t.Fatal("invalid result, result state does not match reference state:", testeeResultState.Diff(referenceResultState), errorReportString(state, testeeResultState, referenceResultState))
```

Comment:
maybe it's a nit thing, but I don't like `invalid result, result state`, maybe `resulting state` or `test result state`. wdyt ?

Reply:
```suggestion
			t.Fatal("invalid result, resulting state does not match reference state:", testeeResultState.Diff(referenceResultState), errorReportString(state, testeeResultState, referenceResultState))
```